File: 14-longest-common-prefix/14-longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
const longestCommonPrefix = function(strs) {
    let prefix = '';
    for (let i = 0; i < strs[0].length; i++) {
        const c = strs[0][i];
        for (const str of strs) {
            if (str[i] !== c) return prefix;
        }
        prefix += c;
    }
    return prefix;
};

File: 14-longest-common-prefix/14-longest-common-prefix.rb
# @param {String[]} strs
# @return {String}
def longest_common_prefix(strs)
    prefix = ''
    strs[0].each_char.with_index do |char, i|
        strs.each { |str| return prefix if str[i] != char}
        prefix += char
    end
end